### PR TITLE
JENA-2031: Fix handling of base URI for Windows file systems.

### PR DESCRIPTION
**Pull request by afs**
This is fixing the build failures on Windows where full file names including the drive letter, in `\` form.
This is a bug in the JENA-2031 changes causing an invalid URI for the base, which in turn lead to an exception.

### CODE REVIEW

File: jena-arq/src/main/java/org/apache/jena/sparql/expr/aggregate/AggMode.java [hunk 1]
```diff
@@ -1,139 +1,139 @@
...
+    	        }
+    	        else {
+    	            amode.put(arrDouble[i],1);
+    	            if(mode==0) {
+    	            	mode = arrDouble[i];
+    	            }
+    	        }
+    	    }
+
+            return NodeValue.makeDecimal(mode);
```
Comment (kinow) [inline jena-arq/src/main/java/org/apache/jena/sparql/expr/aggregate/AggMode.java:132]: Extra spaces

File: jena-arq/src/main/java/org/apache/jena/sparql/expr/aggregate/AggModeDistinct.java [hunk 2]
```diff
@@ -1,138 +1,138 @@
...
+    	        }
+    	        else {
+    	            amode.put(arrDouble[i],1);
+    	            if(mode==0) {
+    	            	mode = arrDouble[i];
+    	            }
+    	        }
+    	    }
+
+            return NodeValue.makeDecimal(mode);
```
Comment (kinow) [inline jena-arq/src/main/java/org/apache/jena/sparql/expr/aggregate/AggModeDistinct.java:131]: Using GH mobile UI. This file has a few lines with extra spaces (or tabs?) I think.

**Review by kinow — APPROVED** [review]

**Comment by kinow** [comment]
Would it be worth adding a unit test (if not yet covered) for the single letter windows partition name? To prevent a regression..

File: jena-arq/src/main/java/org/apache/jena/sparql/expr/aggregate/AggModeDistinct.java (hunk 2 above)
Comment (afs) [reply jena-arq/src/main/java/org/apache/jena/sparql/expr/aggregate/AggModeDistinct.java:131]: Yes - tabs crept in, not at the start of the line - the files  in the PR were converted from windows line mode.

**Comment by afs** [comment]
There is a test in `TestJenaReaderRIOT` - it tries all the ways to write file names.

It was `read_url_3` and `read_url_3x` that were failing on the Jenkins windows build.

**Pull request merged**